pylint/testutils/functional/test_file.py

```diff
@@ -99,7 +101,19 @@ def module(self) -> str:
 
     @property
     def expected_output(self) -> str:
-        return self._file_type(".txt", check_exists=False)
+        files = [
+            p.stem
+            for p in Path(self._directory).glob(f"{split(self.base)[-1]}.[0-9]*.txt")
+        ]
+        # pylint: disable-next=bad-builtin
+        current_version = int("".join(map(str, sys.version_info[:2])))
+        output_options = [
+            int(version) for s in files if (version := s.rpartition(".")[2]).isalnum()
+        ]
+        for opt in sorted(output_options, reverse=True):
+            if current_version >= opt:
```

Comment:
Just noting this will break for a hypothetical python 4, can we immediately convert the strings to tuples and compare those?